### PR TITLE
added segmentvisibility toggle

### LINKED ISSUE #57: Hide segmentation
It should be possible to hide the current segmentation result (keyboard shortcut...) as it can be distracting when manually marking an image etc.
This setting should stay active until the user disables it (shortcut), runs the segmentation or switches to another page.

### PR DESCRIPTION
**Pull request by Kolophonium0**
Closes #57

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag [hunk 1]
```diff
@@ -819,6 +819,18 @@
                             </tr>
                         </table>
                     </li>
+                    <li>
+                        <table>
+                            <tr>
+                                <td class="kbd">
+                                    <span>shift</span>+<span>h</span>
```
Comment (maxnth) [inline src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag:826]: Shortcut key description should follow capitalization standards

File: src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag [hunk 2]
```diff
@@ -819,6 +819,18 @@
...
+                                </td>
+                                <td class="kbdtext">
+                                    toggle visibility of the previous segmentation
```
Comment (maxnth) [inline src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag:829]: Shortcut description should follow capitalization standards of other shortcut descriptions

File: src/main/webapp/resources/js/viewer/gui.js [hunk 3]
```diff
@@ -938,7 +938,18 @@ function GUI(canvas, viewer, colors, accessible_modes) {
 		});
 		
 	}
-	
+
+	this.hideSegments = function (removedChildren) {
```
Comment (maxnth) [inline src/main/webapp/resources/js/viewer/gui.js:942]: unused parameter?

**Comment by maxnth** [comment]
Adding some kind of display (e.g. switch in the sidebar) to inform the user about the current state of the program (in this case whether previous segments are hidden or not) would probably be helpful.

File: src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag (hunk 2 above)
Comment (maxnth) [reply src/main/webapp/WEB-INF/tags/keyboardShortcutModal.tag:829]: Shortcut should also be added to all Tabs where the shortcut is available (in this case `Edit` and `Segment`)

**Comment by maxnth** [comment]
Not to be nit-picky but the toast obstructs the view to one or more elements (depending on the resolution) in the sidebar (regions, reading order, ...) which are only fully accessible after undoing the hiding.
image
image

**Comment by Kolophonium0** [comment]
Yeah I stole this idea from the batch segmentation Toast, and did not solved this in the hope you had some commented css for this somewhere?
________________________________

**Comment by Kolophonium0** [comment]
But the concept works for you?
________________________________

**Comment by maxnth** [comment]
While moving the toast a bit to the right moves it out of the way of the settings it might get in the way of the working canvas while segmenting as you can't click anything under the toast. 
If you accidentally swipe the toast to the right/left you can't undo the hiding (not even via shortcut as the function tries to remove the toast which doesn't exist anymore) and the toast doesn't disappear on page change.

IMO just adding a toggle to the right sidebar (see e.g. toggles under `Settings` for `Reading Order hints` etc.) as already suggested would probably be a lot easier to implement and more consistent with the current UI. 
You'd also be able to activate via mouse (instead of only allowing hiding via shortcut but allowing "unhide" via button in the toast and shortcut).  
Next to adding it somehwere in the sidebar you'd only have to trigger the hide/show function on mouse click and programatically toggle the state on shortcut press (+ obviously reset on actions which reset hide/show like page change).

**Comment by Kolophonium0** [comment]
Not sure about the placing

**Comment by Kolophonium0** [comment]
@maxnth How's it looking?

**Comment by maxnth** [comment]
Toggle doesn't reset on page change or other actions which reset the visibility state...

Peek 2020-09-01 14-19

Please test obvious stuff like this locally before requesting reviews

**Comment by Kolophonium0** [comment]
> Toggle doesn't reset on page change or other actions which reset the visibility state...
> 
> Peek 2020-09-01 14-19
> 
> Please test obvious stuff like this locally before requesting reviews

Thank you i wasn't aware that obviously none of the Settings GUI Part gets reset after page load. Can you tell me the other actions too?

**Comment by maxnth** [comment]
`Undo` and `Redo` force visibility of all segmentation so that users don't remove/change segments by accident without even seeing them/the changes

**Pull request merged**